### DISCUSSION TITLE
Strange traffic selectors after some time on remote access IKEv2

### DISCUSSION THREAD

**Discussion by t-dubbb** [strongswan/strongswan#1591] [Q&A]
I have a remote access IKEv2 Strongswan running on a Ubiquiti Edgerouter Lite that I am connecting to from an iPhone using the phone's built-in VPN client. The config is shown below:

> conn RA
>     keyexchange=ikev2
>     ike=aes128-sha256-modp2048!
>     esp=aes128-sha1-modp2048,aes128-sha1!
>     ikelifetime=86400
>     fragmentation=yes
>     keylife=3600
>     rekey=yes
>     forceencaps=yes
>     dpdaction=restart
>     dpddelay=63s
>     left=%any
>     leftca=/config/auth/ca.crt
>     leftauth=pubkey
>     leftid="SERVER_DNS_NAME"
>     leftsubnet=0.0.0.0/0
>     leftcert=/config/auth/Home.crt
>     leftsendcert=always
>     right=%any
>     rightauth=pubkey
>     rightsourceip=172.16.16.64/29
>     rightdns=172.16.16.2
>     rightid=%any
>     rightca=%same
>     rightsendcert=never
>     #eap_identity=%identity
>     auto=add

At first, the device connects and gets a normal looking SA:

> RA: #5, ESTABLISHED, IKEv2, 5ce537e9cf4fe1c1_i 953a30cae520b351_r*
>   local  'SERVER_DNS_NAME' @ SERVER_WAN_IP[4500]
>   remote 'PHONE_CERT_DN' @ PHONE_WAN_IP[14651] [172.16.16.65]
>   AES_CBC-128/HMAC_SHA2_256_128/PRF_HMAC_SHA2_256/MODP_2048
>   established 8660s ago, reauth in 77135s
>   RA: #26, reqid 4, INSTALLED, TUNNEL-in-UDP, ESP:AES_CBC-128/HMAC_SHA1_96/MODP_2048
>     installed 573s ago, rekeying in 2317s, expires in 3028s
>     in  c2ae9444,  10548 bytes,   161 packets
>     out 005b7b63,      0 bytes,     0 packets
>     local  0.0.0.0/0
>     remote 172.16.16.65/32

But after a while, the "in" SA switches from 0.0.0.0/0 to 172.16.16.65/32........

The logs show this happening:

> .
> .
> .
> Mar 10 19:37:56 12[IKE] <RA|5> IKE_SA RA[5] established between SERVER_WAN_IP[SERVER_DNS_NAME]...PHONE_WAN_IP[PHONE_CERT_DN]
> Mar 10 19:37:56 12[IKE] <RA|5> CHILD_SA RA{18} established with SPIs cbb17634_i 04ce469a_o and TS 0.0.0.0/0 === 172.16.16.65/32
> Mar 10 20:22:05 08[IKE] <RA|5> establishing CHILD_SA RA{20} reqid 4
> Mar 10 20:22:07 15[IKE] <RA|5> inbound CHILD_SA RA{20} established with SPIs c2e6e9e1_i 0ae8d56c_o and TS 0.0.0.0/0 === 172.16.16.65/32
> Mar 10 20:22:07 15[IKE] <RA|5> outbound CHILD_SA RA{20} established with SPIs c2e6e9e1_i 0ae8d56c_o and TS 0.0.0.0/0 === 172.16.16.65/32
> Mar 10 20:22:07 15[IKE] <RA|5> closing CHILD_SA RA{18} with SPIs cbb17634_i (36841233 bytes) 04ce469a_o (176508946 bytes) and TS 0.0.0.0/0 === 172.16.16.65/32
> Mar 10 21:10:09 12[IKE] <RA|5> inbound CHILD_SA RA{24} established with SPIs caac818d_i 0bc88e36_o and TS 172.16.16.65/32 === 172.16.16.65/32
> Mar 10 21:10:09 10[IKE] <RA|5> closing CHILD_SA RA{20} with SPIs c2e6e9e1_i (3265427 bytes) 0ae8d56c_o (68846932 bytes) and TS 0.0.0.0/0 === 172.16.16.65/32
> Mar 10 21:10:09 10[IKE] <RA|5> outbound CHILD_SA RA{24} established with SPIs caac818d_i 0bc88e36_o and TS 172.16.16.65/32 === 172.16.16.65/32
> Mar 10 21:52:43 15[IKE] <RA|5> establishing CHILD_SA RA{26} reqid 4
> Mar 10 21:52:44 08[IKE] <RA|5> inbound CHILD_SA RA{26} established with SPIs c2ae9444_i 005b7b63_o and TS 172.16.16.65/32 === 172.16.16.65/32
> Mar 10 21:52:44 08[IKE] <RA|5> outbound CHILD_SA RA{26} established with SPIs c2ae9444_i 005b7b63_o and TS 172.16.16.65/32 === 172.16.16.65/32
> Mar 10 21:52:44 08[IKE] <RA|5> closing CHILD_SA RA{24} with SPIs caac818d_i (71715 bytes) 0bc88e36_o (0 bytes) and TS 172.16.16.65/32 === 172.16.16.65/32
> .
> .
> .

??? what is happening here? Of course when this happens, the phone loses all routing whatsoever. Even more weirdly, the actual SA stays up, with a packet sniff showing DPD messages continuing to be exchanged. Any ideas what's going on?

**Comment by tobiasbrunner** [comment]
No idea. Try increasing the log level (see #196 for a snippet with recommended levels). What strongSwan version is the Ubiquiti box running?
